File: 7_ReverseInteger/7_v1.java
public class Solution {
    public int reverse(int x) {
        StringBuilder stringValue = new StringBuilder(Integer.toString(Math.abs(x)));
        String reverseStr = stringValue.reverse().toString();
        int result = 0;
        
        try {
            result = Integer.parseInt(reverseStr);
        } catch(Exception e) {
        }
        
        return result * (int)Math.signum(x);
    }
}

File: 7_ReverseInteger/7_v1.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        stringValue = str(abs(x))
        reverseStr = stringValue[::-1]
        
        if x > 0 and int(reverseStr) <= 2147483647:
            return int(reverseStr)
        elif x < 0 and -int(reverseStr) >= -2147483648:
            return -int(reverseStr)
        
        return 0
